modules/transactions/src/test/java/org/apache/ignite/internal/tx/TxManagerTest.java

```diff
@@ -343,7 +343,10 @@ void testRepeatedCommitRollbackAfterCommitWithException() throws Exception {
 
         TransactionException transactionException = assertThrows(TransactionException.class, tx::commit);
 
-        assertInstanceOf(MismatchingTransactionOutcomeException.class, ExceptionUtils.unwrapCause(transactionException.getCause()));
+        assertInstanceOf(
+                MismatchingTransactionOutcomeException.class,
+                ExceptionUtils.unwrapCause(transactionException.getCause().getCause().getCause())
```

Comment:
Perhaps, using `IgniteTestUtils.hasCause` would be more readable.

Reply:
Changed to `IgniteTestUtils.hasCause()`